UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.FulfilmentService/Services/FulfilmentDataService.cs

```diff
@@ -425,18 +426,21 @@ private async Task<bool> CreateLargeMediaSerialEncFile(string batchId, string ex
                       "Create large media serial enc file request for BatchId:{batchId} and _X-Correlation-ID:{CorrelationId}",
                       async () =>
                       {
-                          var baseDirectorys = fileSystemHelper.GetDirectoryInfo(Path.Combine(exchangeSetPath, string.Format(rootFolderName, rootFolderNumber)))
-                                                  .Where(di => di.Name.StartsWith("B") && di.Name.Count() == 2 && char.IsDigit(Convert.ToChar(di.Name.ToString().Substring(di.Name.ToString().Length - 1))));
+                          var baseDirectoryies = fileSystemHelper.GetDirectoryInfo(Path.Combine(exchangeSetPath, rootfolder))
+                                                  .Where(di => di.Name.StartsWith("B") && di.Name.Count() <= 3 && char.IsDigit(Convert.ToChar(di.Name.ToString()[^1..])));
```

Comment:
Please check this logic again with B10. 

di.Name looks like string only no need to convert again please check other places as well.